crates/biome_js_analyze/src/analyzers/nursery/use_shorthand_assign.rs

```diff
@@ -0,0 +1,268 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, Rule, RuleDiagnostic,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsExpression, JsAssignmentExpression, JsAssignmentOperator, JsBinaryExpression,
+    JsBinaryOperator, JsSyntaxKind, T,
+};
+use biome_rowan::{AstNode, BatchMutationExt};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Require assignment operator shorthand where possible
+    ///
+    /// JavaScript provides shorthand operators that combine variable assignment and some simple mathematical operations
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/operator-assignment/
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
```

Comment:
Usually, we try to put only a few examples explaining how the rule works in the documentation. I believe 3/4 should be enough (3 valid and 3 invalid)